elements/rh-dialog/rh-dialog.css

```diff
@@ -1,194 +1,254 @@
+  inline-size: 100%;
 }
 
-[part='dialog'] {
-  position: relative;
-  margin: 0 auto;
-  width: var(--_box-width, calc(100% - var(--rh-space-2xl, 32px)));
-  max-height: var(--_box-max-height, calc(100% - var(--rh-space-3xl, 48px)));
-  box-shadow:
-    0 1rem 2rem 0 rgba(3, 3, 3, 0.16),
-    0 0 0.5rem 0 rgba(3, 3, 3, 0.1);
-  padding: var(--rh-space-xl, 24px);
-  margin-inline: var(--rh-space-lg, 16px);
-  background-color: var(--rh-color-surface-lightest, #ffffff);
-  max-width: min(90%, 1140px);
-  border-radius: var(--rh-border-radius-default, 3px);
-  color: var(--rh-color-text-primary-on-light, #151515);
-  font-family: inherit;
+::backdrop {
+  background-color: var(--rh-dialog-backdrop-background-color);
 }
 
-:host([width]) [part='dialog'],
-:host([variant]) [part='dialog'] {
-  margin-inline: 0;
+#dialog {
+  background-color: var(--rh-color-surface);
+  border-radius: var(--rh-border-radius-default, 3px);
+  border: 0;
+  box-shadow: var(--rh-box-shadow-xl, 0 8px 24px 3px rgba(21, 21, 21, 0.35));
+  box-sizing: border-box;
+  color: var(--rh-color-text-primary);
+  font-family: inherit;
+  inline-size: 100%;
+  margin-inline: auto;
+  max-block-size: var(--_box-max-block-size, calc(100vh - var(--rh-space-3xl, 48px)));
+  max-inline-size: var(--_box-width, min(90%, 1140px));
+  overflow-y: auto;
+  overscroll-behavior: contain;
+  padding: var(--rh-space-xl, 24px);
+  position: relative;
 }
 
-:host([width='small']) [part='dialog'],
-:host([variant='small']) [part='dialog'] {
+:host([width='small']) #dialog,
+:host([variant='small']) #dialog {
   --_box-width: 35rem;
 }
 
-:host([width='medium']) [part='dialog'],
-:host([variant='medium']) [part='dialog'] {
+:host([width='medium']) #dialog,
+:host([variant='medium']) #dialog {
   --_box-width: 52.5rem;
 }
 
-:host([width='large']) [part='dialog'],
-:host([variant='large']) [part='dialog'] {
+:host([width='large']) #dialog,
+:host([variant='large']) #dialog {
   --_box-width: 70rem;
 }
 
-[part='content'] {
-  overflow-y: auto;
-  overscroll-behavior: contain;
-  max-height: var(--_box-max-height, calc(100vh - var(--rh-space-3xl, 48px)));
-  box-sizing: border-box;
-  border-radius: var(--rh-border-radius-default, 3px);
+#content {
+  margin-block-start: calc(-1 * var(--offset-top));
 }
 
-[part='content'] ::slotted([slot='header']) {
-  margin-top: 0 !important;
+#header {
+  background-color: var(--rh-color-surface);
+  inset-block-start: 0;
+  margin-block-end: var(--rh-space-lg, 16px);
+  position: relative;
+
+  &:before {
+    background-color: var(--rh-color-surface);
+    block-size: 100%;
+    content: '';
+    inline-size: 100%;
+    inset-block-start: calc(-1 * var(--offset-top));
+    position: absolute;
+    z-index: -1;
+  }
+
+  &:after {
+    background-color: var(--rh-color-surface);
+    block-size: var(--rh-space-lg, 16px);
+    content: '';
+    inline-size: 100%;
+    position: absolute;
+    z-index: -1;
+  }
+
+  @media (min-height: 768px) {
```

Comment:
please leave a comment with the name of the breakpoint token e.g. Please also use the new range syntax

```css
/* --rh-media-sm */
@media (height >= 768px) {
```